### PR TITLE
kapacitor threshold critical value strings

### PR DESCRIPTION
**Pull request by goller**
- [x] CHANGELOG.md updated
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #865 

### The problem
If threshold alert rule's critical value was a string, the tickscript wasn't generated correctly.

### The Solution
This checks if the critical value is a string and formats it with kapacitor's single-quotes.

This patch does not address string inequality nor threshold ranges.

### CODE REVIEW

File: kapacitor/tickscripts_test.go [hunk 1]
```diff
@@ -199,6 +199,280 @@ trigger
...
+			Database:        "influxdb",
+			RetentionPolicy: "autogen",
+			Measurement:     "haproxy",
+			Fields: []chronograf.Field{
+				{
+					Field: "status",
+					Funcs: []string{"last"},
+				},
+			},
+			GroupBy: chronograf.GroupBy{
```
Comment (timraymond) [inline kapacitor/tickscripts_test.go]: Should all of these AST nodes live under the `kapacitor` package? I know it wasn't introduced in this patch so we can make the change in another PR if you feel it's worthwhile.

File: kapacitor/vars.go [hunk 2]
```diff
@@ -39,15 +40,16 @@ func Vars(rule chronograf.AlertRule) (string, error) {
 		%s
         var crit = %s
  `
-			return fmt.Sprintf(vars,
-				common,
-				rule.TriggerValues.Value), nil
+			// If critical value is a string, we'll
+			// need to single-quote it.
+			crit := critVar(rule.TriggerValues.Value)
```
Comment (timraymond) [inline kapacitor/vars.go]: maybe `critVar` could be named `quote` or `formatQuote` or something? I feel like the comment above wants a better name out of this function :) You could potentially inline it too with a different name:

```go
return fmt.Sprintf(vars, common, formatQuote(rule.TriggerValues.Value))
```

File: kapacitor/tickscripts_test.go (hunk 1 above)
Comment (goller) [inline kapacitor/tickscripts_test.go]: In the long run yes, I think you are right. Currently, the kapacitor package is translating from the front-ends QueryConfig whereas I think in the future we won't use the QueryConfig to create the rule;  I think in the future we'll use something much closer to kapacitor's AST.

File: kapacitor/vars.go (hunk 2 above)
Comment (goller) [inline kapacitor/vars.go]: Good point.  Fixed in febc93a2

(I went with `formatValue`)
